### ISSUE TITLE
SslProtocols.Tls13: TLS Handshake Header Lists TLS 1.0

### ISSUE THREAD

**Issue by felsokning** [dotnet/runtime#99071]
### Description

When using TLS 1.3 (`SslProtocols.Tls13`), it appears that the TLS Handshake Header is defaulting to TLS 1.0. This causes servers that are not configured to accept TLS 1.0 to send an ack-reset in response. This reproduces in .NET5 (I know, I know, OOS -- out of service/support), .NET6, .NET7, and .NET8.

### Reproduction Steps

            var httpClientHandler = new HttpClientHandler()
            {
                SslProtocols = SslProtocols.Tls13
            };

            var httpClient = new HttpClient(httpClientHandler);
            var result = await httpClient.GetStringAsync("https://api.nationaltransport.ie/gtfsr/v2/TripUpdates?format=json");

### Expected behavior

The handshake header should have the matching TLS version configured (e.g.: 1.3).

### Actual behavior

The handshake header has TLS 1.0 defined.

### Regression?

Unknown.

### Known Workarounds

Using TLS 1.2 (`SslProtocols = SslProtocols.Tls12`) is the only known work-around, at the moment.

### Configuration

Versions: .NET5, .NET6, .NET7, .NET8
OS: Windows 11 Professional (23H2 22631.3155)
Architecture: x64
Specific to this configuration: Unknown

### Other information

I opened a [bug in Developer Community](https://developercommunity.visualstudio.com/t/SslProtocolsTls13:-Handshake-Header-Lis/10603930) for this -- but that was clearly the wrong place for it.

**Comment by wfurt** [comment]
This is expected and part of normal operation (for compat) You can look for example at 
` openssl s_client -connect www.github.com:443 -tls1_3`

![image](https://github.com/dotnet/runtime/assets/14356188/aee48e5b-adbc-45a7-a9a6-6311de1d2379)

The header is irrelevant. what matters are `supporeted_versions` and negotiated versions. 

I think what causes the RST is lack of TLS1.3 support. I you feel otherwise please prove more details. 
And perhaps scout that site with https://www.ssllabs.com/ssltest/

**Comment by felsokning** [comment]
So, if I understand this - correctly - TLS 1.2 will always send TLS 1.2 version in the handshake header but TLS 1.3 will always send TLS 1.0 in the handshake header? So, TLS 1.2 is _not_ backwards compatible - but TLS 1.3 is?

![image](https://github.com/dotnet/runtime/assets/46375514/7b9b4a4f-8765-4687-8d60-cbca9ab04782)

**Comment by wfurt** [comment]
just different https://datatracker.ietf.org/doc/html/rfc8446#section-1.2
```
     The TLS 1.2 version negotiation mechanism has been deprecated in
      favor of a version list in an extension.  This increases
      compatibility with existing servers that incorrectly implemented
      version negotiation.
```

**Comment by felsokning** [comment]
Yes, I kept getting socket closed by remote host, no matter what version of .NET I tried.

![image](https://github.com/dotnet/runtime/assets/46375514/e7593476-89d2-4172-bdc2-69072e8b94ec)

I tried contacting the site owners, who took one look at the handshake header and decried me using TLS 1.0. 

I use an inherited class and had forgotten that [I had set TLS 1.3 on the handler](https://github.com/felsokning/Fels-kning/blob/06f7ee8595b3c30b709dc7a9190bc5476f58f33a/Fels%C3%B6kning/HttpBase.cs#L23). Taking the site's owners words as a sort of gospel (I'm no rocket-scientist at networking, to be sure, I opened this bug).

Now that I know/understand the behavior, I'll go back and tell them the fix for others running into the issue: 

Either turn on TLS 1.3 or tell people to _only_ use TLS 1.2.
